core/clustersmngr/factory_caches.go

```diff
@@ -8,19 +8,25 @@ import (
 
 	"github.com/cheshir/ttlcache"
 	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+	"github.com/weaveworks/weave-gitops/pkg/server/sets"
 	v1 "k8s.io/api/core/v1"
 )
 
 type Clusters struct {
 	sync.RWMutex
-	clusters []Cluster
+	clusters sets.New[Cluster]
 }
 
-func (c *Clusters) Set(clusters []Cluster) {
+func (c *Clusters) Set(clusters sets.New[Cluster]) (sets.New[Cluster], sets.New[Cluster]) {
```

Comment:
Lets keep the public interface the same for now while exploring this

```suggestion
func (c *Clusters) Set(clustersList []Cluster) ([]Cluster, []Cluster) {
```

And create the `set.New(clustersList)` in here and return `added.List(), removed.List()` etc